src/Metadata/Cip25/V2.purs

```diff
@@ -0,0 +1,348 @@
+-- | ```
+newtype Cip25MetadataEntry = Cip25MetadataEntry
+  { policyId :: MintingPolicyHash
+  , assetName :: Cip25TokenName
+  -- metadata_details:
+  , name :: Cip25String
+  , image :: String
+  , mediaType :: Maybe Cip25String
+  , description :: Maybe String
+  , files :: Array Cip25MetadataFile
+  }
+
+derive instance Generic Cip25MetadataEntry _
+derive instance Newtype Cip25MetadataEntry _
+derive instance Eq Cip25MetadataEntry
+
+instance Show Cip25MetadataEntry where
+  show = genericShow
+
+-- | A version tag (not exported, internal)
+data Cip25V2 = Cip25V2
+
+derive instance Eq Cip25V2
+derive instance Ord Cip25V2
+derive instance Generic Cip25V2 _
+
+instance Show Cip25V2 where
+  show = genericShow
+
+instance FromData Cip25V2 where
+  fromData (Integer int)
+    | int == BigInt.fromInt 2 = pure Cip25V2
+  fromData _ = Nothing
+
+instance ToData Cip25V2 where
+  toData _ = Integer $ BigInt.fromInt 2
+
+instance FromMetadata Cip25V2 where
+  fromMetadata (Int int)
+    | Int.toBigInt int == BigInt.fromInt 2 = pure Cip25V2
+  fromMetadata _ = Nothing
+
+instance ToMetadata Cip25V2 where
+  toMetadata _ = Int $ unsafePartial $ fromJust $ Int.fromBigInt $
+    BigInt.fromInt 2
+
+instance DecodeAeson Cip25V2 where
+  decodeAeson aeson = do
+    n :: Int <- decodeAeson aeson
+    case n of
+      2 -> pure Cip25V2
+      _ -> Left $ TypeMismatch "Cip25V2"
+
+metadataEntryToMetadata :: Cip25MetadataEntry -> TransactionMetadatum
+metadataEntryToMetadata (Cip25MetadataEntry entry) = toMetadata $
+  [ "name" /\ anyToMetadata entry.name
+  , "image" /\ anyToMetadata entry.image
+  ]
+    <> mbMediaType
+    <> mbDescription
+    <> mbFiles
+  where
+  mbFiles = case entry.files of
+    [] -> []
+    files -> [ "files" /\ anyToMetadata files ]
+  mbMediaType = fold $ entry.mediaType <#> \mediaType ->
+    [ "mediaType" /\ anyToMetadata mediaType ]
+  mbDescription = fold $ entry.description <#> \description ->
+    [ "description" /\ anyToMetadata description ]
+
+metadataEntryFromMetadata
+  :: MintingPolicyHash
+  -> Cip25TokenName
+  -> TransactionMetadatum
+  -> Maybe Cip25MetadataEntry
+metadataEntryFromMetadata policyId assetName contents = do
+  name <- lookupMetadata "name" contents >>= fromMetadata
+  image <- lookupMetadata "image" contents >>= fromMetadataString
+  mediaType <- for (lookupMetadata "mediaType" contents) fromMetadata
+  description <- for (lookupMetadata "description" contents) fromMetadataString
+  files <- for (lookupMetadata "files" contents) fromMetadata <#> fromMaybe []
+  pure $
+    wrap { policyId, assetName, name, image, mediaType, description, files }
+
+metadataEntryToData :: Cip25MetadataEntry -> PlutusData
+metadataEntryToData (Cip25MetadataEntry entry) = toData $ AssocMap.Map $
+  [ "name" /\ toData entry.name
+  , "image" /\ toData entry.image
+  ]
+    <> mbMediaType
+    <> mbDescription
+    <> mbFiles
+  where
+  mbFiles = case entry.files of
+    [] -> []
+    files -> [ "files" /\ toData files ]
```

Comment:
I think this case can be done the same as the others, right?

Reply:
It's Array instead of Maybe, so no